runtime/hsa-runtime/core/util/utils.h

```diff
@@ -394,6 +394,39 @@ inline uint32_t PtrHigh32(const void* p) {
   return ptr;
 }
 
+/// @brief: Concatenate two 32-bit numbers into a 64-bit number
+/// @param: highAddr(Input), To be put in the upper 32-bits of the output
+/// @param: lowAddr(Input), To be put in the lower 32-bits of the output
+/// @return: uint64_t, Concatenation of high and low
+inline uint64_t ConcatAddr(const uint32_t high, const uint32_t low) {
```

Comment:
This could be useful to generalize to arbitrary sizes. For example, this should work:

```c++
template <typename OutType, typename InType,
          std::enable_if_t<std::is_integral<OutType>::value &&
          std::is_integral<InType>::value && sizeof(OutType) >= 2 * sizeof(InType), bool> = true>
inline OutType Concat(InType hi, InType lo) {
  OutType res = ((static_cast<OutType>(hi) << sizeof(InType) * 8) | static_cast<OutType>(lo));
  return res;
}
```